mlkaps/clustering/clustering_visualization.py

```diff
@@ -153,28 +154,27 @@ def _set_axis_range(ax, configuration):
     input_parameters = configuration.input_parameters
 
     ax.set_xlabel(input_parameters[0])
-    ax.set_xlim(features_values[input_parameters[0]][0], features_values[input_parameters[0]][1])
+    ax.set_xlim(*features_values[input_parameters[0]].get_sampling_bounds())
 
     if len(input_parameters) == 2:
         ax.set_ylabel(input_parameters[1])
-        ax.set_ylim(
-            features_values[input_parameters[1]][0],
-            features_values[input_parameters[1]][1],
-        )
+        ax.set_ylim(*features_values[input_parameters[1]].get_sampling_bounds())
     elif len(input_parameters) == 1:
         ax.set_ylabel(configuration["parameters"]["design"][0])
         plt.legend()
 
 
 def _scatter_optimization_results(ax, color_map, configuration, design_param, optimization_results):
-    feature_values = configuration["parameters"]["features_values"][design_param]
+    features = configuration["parameters"]["features_values"]
+    feature_values = features[design_param]
     y = optimization_results[design_param]
     values_y = np.unique(y)
 
     # Plot the optimization results on the same plot
-    feature_type = configuration.parameters_type[design_param]
     input_parameters = configuration.input_parameters
-    if feature_type in ["Categorical", "Boolean"]:
+    if isinstance(feature_values, ValueSet):
+        # FIXME: this was broken even before ValueContainers
+        # inverse_value_map = {v: k for k, v in enumerate(feature_values.values)} ????
```

Comment:
@Thukisdo not sure what was the point here, but since it is part of vizualization it is not critical to fix. Worst you can open an issue to look at it later.

Reply:
same let's open an issue here